.github/workflows/go.yml

```diff
@@ -16,8 +16,57 @@ jobs:
         uses: actions/checkout@v2
 
       - name: Install Sphincsplus dependency
-        run: ./.github/scripts/install_sphincs.sh
-        shell: bash
+        run: cd sphincsplus/ref && make libsphincsplus.so && sudo make install && sudo ldconfig
 
       - name: Run all package unit tests
-        run: sudo sh -c "ulimit -l 64435 && GORACE=history_size=7 go test -v -timeout 0 ./..."
+        run: sudo sh -c "ulimit -l 64435 && GORACE=history_size=7 go test -race -v -timeout 0 ./..."
+  test_e2e:
+    env:
+      warped: true
+    strategy:
+      matrix:
+        go-version: [1.17.x]
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install Sphincsplus dependency
+        run: cd sphincsplus/ref && make libsphincsplus.so && sudo make install && sudo ldconfig
```

Comment:
main branch has the install_sphincs.sh script which now does this properly with static linking unlike your line here. My suggestion is to remove this little section and not to remove the shell script... that way you aren't undoing my work.

Reply:
ah I removed it because your script was cloning the whole katzenpost repository and wrong.